returnn/tf/layers/basic.py

```diff
@@ -607,6 +607,59 @@ def get_out_data_from_opts(cls, sources, name, **kwargs):
     return get_concat_sources_data_template(sources, name="%s_output" % name)
 
 
+class NormLayer(_ConcatInputLayer):
+  """
+  Normalize over specified axes, e.g. time and/or feature axis.
+  In case of just feature, this corresponds to layer norm.
+  """
+  layer_class = "norm"
+
+  def __init__(self, axes, param_shape="F", scale=True, bias=True, epsilon=1e-6, **kwargs):
+    """
+    :param str|list[str] axes: axes over which the mean and variance are computed, e.g. "F" or "TF"
+    :param str|list[str]|int|list[int] param_shape: shape of the scale and bias parameters
+    :param bool scale: add trainable scale parameters
+    :param bool bias: add trainable bias parameters
+    :param float epsilon: epsilon for numerical stability
+    """
+    super(NormLayer, self).__init__(**kwargs)
+    assert not self.input_data.sparse
+    x = self.input_data.placeholder
+    dim = []
+    if scale or bias:
+      if isinstance(param_shape, int):
+        dim = [param_shape]
+      elif isinstance(param_shape, list) and all(isinstance(x, int) for x in param_shape):
+        dim = param_shape
+      elif isinstance(param_shape, str) or (
+           isinstance(param_shape, list) and all(isinstance(x, int) for x in param_shape)):
```

Comment:
The check is wrong. You want that all x are str.

Reply:
Also, indentation is wrong.

Reply:
>You want that all x are str.
Yes, definitely.

>Also, indentation is wrong.
What level of indentation would you prefer? 2 spaces compared to elif is the same as the next line, which is not nice. 4 spaces triggers a PyCharm warning regarding PEP8.

Reply:
Not really sure on indentation. Even two more (because last line ended with `(`). Or maybe three more (such that it is a multiple of 2 in total).
Or maybe directly use `elif (` and then newline, with 4 or 6 spaces indent.